api/server/router/container/container_routes.go

```diff
@@ -332,11 +333,17 @@ func (s *containerRouter) postContainerUpdate(ctx context.Context, w http.Respon
 		return err
 	}
 
-	_, hostConfig, err := runconfig.DecodeContainerConfig(r.Body)
-	if err != nil {
+	var updateConfig container.UpdateConfig
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&updateConfig); err != nil {
 		return err
 	}
 
+	hostConfig := &container.HostConfig{
```

Comment:
Is there further work to do here to get rid of hostConfig? If we end up in a hostConfig anyway, are we really winning anything here?